app/uk/gov/hmrc/gform/eval/smartstring/SmartStringEvaluatorFactory.scala

```diff
@@ -84,11 +84,28 @@ class RealSmartStringEvaluatorFactory() extends SmartStringEvaluatorFactory {
             s.interpolations
               .map { interpolation =>
                 val typeInfo: TypeInfo = formModelVisibilityOptics.formModel.toFirstOperandTypeInfo(interpolation)
-                val interpolated = formModelVisibilityOptics.evalAndApplyTypeInfo(typeInfo).stringRepresentation
+                def stringRepresentation = formModelVisibilityOptics.evalAndApplyTypeInfo(typeInfo).stringRepresentation
+
+                val interpolated = typeInfo.isChoiceSelection.fold(stringRepresentation) {
+                  case FormCtx(formComponentId) =>
+                    formModelVisibilityOptics.formModel.fcLookup(formComponentId) match {
+                      case IsChoice(choice) =>
```

Comment:
Let support RevealingChoice as well

```
case IsRevealingChoice(rc) => ...
```

Reply:
The ticket says no need to support multi choice fields. Are you talking with revealing choice, with multi = false, or all types of revealing choice in general?